scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -412,3 +413,24 @@ impl DetailedTomlDependency {
         })
     }
 }
+
+/// Merge two `toml::Value` serializable structs.
+pub fn toml_merge<'de, T, S>(target: &mut T, source: &S) -> Result<()>
+where
+    T: Clone + Serialize + Deserialize<'de>,
+    S: Serialize + Deserialize<'de>,
+{
+    let mut params = toml::Value::try_from(target.clone())?;
```

Comment:
```suggestion
/// Merge two `toml::Value` serializable structs.
pub fn toml_merge<'de, T, S>(target: &mut T, source: &S) -> Result<()>
where
    T: Serialize + Deserialize<'de>,
    S: Serialize + Deserialize<'de>,
{
    let mut params = toml::Value::try_from(target)?;
```

https://docs.rs/serde/latest/serde/ser/trait.Serialize.html#impl-Serialize-for-%26'a+mut+T